R/other.R

```diff
@@ -107,13 +107,7 @@ step_other <-
            objects = NULL,
            skip = FALSE,
            id = rand_id("other")) {
-    if (!is_tune(threshold)) {
```

Comment:
There's a lot of moving things here and in #1387. My rationale is that, if a the tuning process inserts a value, we would not check it. For example, `finalize_recipe()` does use the constructor. 

It does defer the checking until prep-time which is later than this but it seems like the only way to really be sure that we are getting corrrect inputs.